src/network/network.cpp

```diff
@@ -0,0 +1,356 @@
+#include "network.h"
+#include "../util/util.h"
+#include "../game.h"
+#include "../test.h"
+
+#include <errno.h>
+#include <cstring>
+#include "imgui/imgui.h"
+
+void NetworkHandle::send(u8 *data, u32 data_len) {
+    int n = write(sockfd, data, data_len);
+    if (n < 0) {
+        ERR("Error writing to socket, errno: {}", errno);
+    } else if ((u32) n < data_len) {
+        ERR("write did not write all data to socket, n={}, data_len={}", n, data_len);
+    }
+}
+
+void NetworkHandle::send(Package *package) {
+    u8 buf[sizeof(Package)];
+    package->id = next_package_id++;
+    pack(buf, package);
+    send(buf, sizeof(Package));
+}
+
+void NetworkHandle::close() {
+    active = false;
+    shutdown(sockfd, SHUT_RDWR);
+    SDL_WaitThread(thread, NULL);
+    ::close(sockfd);
+}
```

Comment:
This should be called from a "shutdown" code piece

Reply:
We don't really have this right now and I think it should be made together with the pre-/post-reload.

Reply:
My bad, we actually have this. Will fix.

Reply:
Wait no we don't, we will soon though (#143).

Note to self: don't code at 1 AM.